src/App.jsx

```diff
@@ -14,33 +15,109 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isAlphabetical, setIsAlphabetical] = useState(false);
+  const [isByLength, setIsByLength] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  // Створюємо ref для кнопки Reverse
+  const reverseButtonRef = useRef(null);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort();
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setIsAlphabetical(true);
+    setIsByLength(false);
+    setIsReversed(false);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (a, b) => a.length - b.length,
+    );
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setIsByLength(true);
+    setIsAlphabetical(false);
+    setIsReversed(false);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(!isReversed);
+
+    // Видаляємо або додаємо клас `is-light` вручну
+    if (reverseButtonRef.current) {
+      if (isReversed) {
+        reverseButtonRef.current.classList.add('is-light');
+      } else {
+        reverseButtonRef.current.classList.remove('is-light');
+      }
+    }
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsAlphabetical(false);
+    setIsByLength(false);
+    setIsReversed(false);
+
+    // Повертаємо клас `is-light` до кнопки Reset
+    if (reverseButtonRef.current) {
+      reverseButtonRef.current.classList.add('is-light');
```

Comment:
Resetting the `is-light` class on the reverse button in `resetGoods` might not be necessary if it is already handled correctly in `reverseGoods`. Ensure that the logic for class toggling is consistent across both functions.